### PR TITLE
posix: Launch gdbserver if thread is killed by signal

### PR DESCRIPTION
**Pull request by Dennisbonke**

### CODE REVIEW

**Review by no92 — APPROVED** [review]
LGTM

File: posix/subsystem/src/process.cpp [hunk 1]
```diff
@@ -5,12 +5,15 @@
...
+#include "gdbserver.hpp"
 #include "process.hpp"
 
 #include <protocols/posix/data.hpp>
 
+#include "debug-options.hpp"
+
 static bool logFileAttach = false;
-static bool logCleanup = false;
+static bool logCleanup_ = false;
```
Comment (Geertiebear) [inline posix/subsystem/src/process.cpp:16]: Why is this change necessary?

Comment (ElectrodeYT) [reply posix/subsystem/src/process.cpp:16]: Conflicts with `debug-options.hpp`

Comment (ikbenlike) [reply posix/subsystem/src/process.cpp:16]: I feel it would be clearer to instead rename it `processLogCleanup` or something along those lines. Other than that, looks fine to me.
